### DISCUSSION TITLE
Troubles with entrez direct

### DISCUSSION THREAD

**Discussion by ialbert** [biostars/biostar-handbook-issues#249] [General]
Entrez direct tools automatically detect the platforms at runtime.

On Mac M1 processors is possible to end up with a situation where the tool is installed as a Rosetta Unix mode but the terminal is running as ARM 1 or vice-versa. 

Reinstall entrez direct with the following command:

    yes | sh -c "$(curl -fsSL ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/install-edirect.sh)" > /dev/null

**Comment by chuddy-ibk** [comment]
I have issues with edirect and since a complete new environment setup also did not work to set up an environment, I try to "repair" the old one where everything else hopefully works. 
when i type:
"
$ esearch -db SSR -query SSR1553606 | head
"

following appears:
" 
WARNING:  FAILURE ( Thu Mar  9 10:25:06 CET 2023 )
nquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ esearch.fcgi -retmax 0 -usehistory y -db SSR -term SSR1553606 -tool edirect -edirect 16.2 -edirect_os Linux -email chuddy@chuddy-HP-ProBook-470-G5
<ERROR>Invalid db name specified: SSR</ERROR>
SECOND ATTEMPT
 WARNING:  FAILURE ( Thu Mar  9 10:25:08 CET 2023 )
nquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ esearch.fcgi -retmax 0 -usehistory y -db SSR -term SSR1553606 -tool edirect -edirect 16.2 -edirect_os Linux -email chuddy@chuddy-HP-ProBook-470-G5
<ERROR>Invalid db name specified: SSR</ERROR>
LAST ATTEMPT
 ERROR:  FAILURE ( Thu Mar  9 10:25:11 CET 2023 )
nquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ esearch.fcgi -retmax 0 -usehistory y -db SSR -term SSR1553606 -tool edirect -edirect 16.2 -edirect_os Linux -email chuddy@chuddy-HP-ProBook-470-G5
<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE eSearchResult>
<eSearchResult>
  <ERROR>Invalid db name specified: SSR</ERROR>
</eSearchResult>
QUERY FAILURE
<ENTREZ_DIRECT>
  <Db>SSR</Db>
  <Step>1</Step>
</ENTREZ_DIRECT>
(bioinfo)
"

When I try to use the 
"yes | sh -c "$(curl -fsSL ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/install-edirect.sh)" > /dev/null"
then following appears: 

"
tar: Skipping to next header

gzip: edirect.tar.gz: invalid compressed data--format violated
tar: Exiting with failure status due to previous errors

gzip: xtract.Linux.gz: invalid compressed data--format violated
chmod: cannot access 'xtract.Linux': No such file or directory

gzip: rchive.Linux.gz: invalid compressed data--format violated
chmod: cannot access 'rchive.Linux': No such file or directory

gzip: transmute.Linux.gz: invalid compressed data--format violated
chmod: cannot access 'transmute.Linux': No such file or directory
(bioinfo)
"

any suggestions?

**Comment by VladimirVinarsky** [comment]
Hello I got the following output from the doctor.py, what should I do?
Thank you for your help

$ doctor.py
# Doctor! Doctor! Give me the news.
# Checking symptoms ...
# bwa           ... OK
# datamash      ... OK
# fastqc -h     ... OK
# hisat2        ... OK
# seqret -h     --> ERROR
# featureCounts ... OK
# efetch        ... OK
# esearch       ... OK
# samtools      ... OK
# fastq-dump    ... OK
# bowtie2       ... OK
# bcftools      ... OK
# seqtk         ... OK
# seqkit        ... OK
# bio           ... OK
# fastq-dump -X 1 -Z SRR1553591 ... OK
#
# Found some problems
#
# Unable to run: seqret -h
#
# Check the Biostar Handbook for installation help.
#
(bioinfo)

**Reply by ialbert** [reply]
it is unexpected that only that tool failed, but in reality that tool is not needed anymore since we have the bio package, so I would recommend moving on

I will redesign the install to not check for this particular tool: seqret

**Reply by VladimirVinarsky** [reply]
Thank you:)
